fca/src/main/java/org/nmdp/ngs/fca/ArbitraryBinaryRelation.java

```diff
@@ -0,0 +1,84 @@
+/*
+
+    ngs-fca  Formal concept analysis for genomics.
+    Copyright (c) 2014-2015 National Marrow Donor Program (NMDP)
+
+    This library is free software; you can redistribute it and/or modify it
+    under the terms of the GNU Lesser General Public License as published
+    by the Free Software Foundation; either version 3 of the License, or (at
+    your option) any later version.
+
+    This library is distributed in the hope that it will be useful, but WITHOUT
+    ANY WARRANTY; with out even the implied warranty of MERCHANTABILITY or
+    FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public
+    License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this library;  if not, write to the Free Software Foundation,
+    Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA.
+
+    > http://www.gnu.org/licenses/lgpl.html
+
+*/
+package org.nmdp.ngs.fca;
+
+import java.util.List;
+import java.util.Map;
+
+public final class ArbitraryBinaryRelation<G extends Relatable, M extends Relatable> {
+    private final List<Entry<G, M>> list;
+    
+    public static class Entry<G, M> implements Map.Entry<G, M> {
```

Comment:
Does this need to be public?  why implement Map.Entry?  could it be immutable (remove setValue method)?